eos-updater-flatpak-installer/main.c

```diff
@@ -0,0 +1,316 @@
+                                     "kind", flatpak_ref_get_kind (flatpak_ref),
+                                     NULL));
+    }
+
+  return g_steal_pointer (&flatpaks_to_deploy);
+}
+
+static GPtrArray *
+get_flatpaks_to_deploy (GFile *to_deploy_file_path, GError **error)
+{
+  g_autofree gchar *contents = NULL;
+
+  if (!g_file_get_contents (g_file_get_path (to_deploy_file_path),
+                            &contents,
+                            NULL,
+                            error))
+    return NULL;
+
+  return parse_flatpaks_to_deploy_from_contents (contents, error);
+}
+
+static const gchar *
+pending_flatpak_deployments_state_path (void)
+{
+  return eos_updater_get_envvar_or ("EOS_UPDATER_TEST_UPDATER_FLATPAK_UPGRADE_STATE_DIR",
+                                    LOCALSTATEDIR "/lib/eos-updater");
+}
+
+static char *
+format_flatpaks (GPtrArray *flatpaks_to_deploy)
+{
+  g_auto(GStrv) flatpaks_to_deploy_strv = g_new0 (gchar *, flatpaks_to_deploy->len + 1);
+  gsize i = 0;
+
+  for (; i < flatpaks_to_deploy->len; ++i)
+    flatpaks_to_deploy_strv[i] = g_strdup (flatpak_ref_get_name (FLATPAK_REF (g_ptr_array_index (flatpaks_to_deploy, i))));
+
+  return g_strjoinv (", ", flatpaks_to_deploy_strv);
+}
+
+static FlatpakInstallation *
+get_flatpak_installation (GError **error)
+{
+  const gchar *override_path = g_getenv ("EOS_UPDATER_TEST_FLATPAK_INSTALLATION_DIR");
+
+  if (override_path)
+    {
+      g_autoptr(GFile) override = g_file_new_for_path (override_path);
+      return flatpak_installation_new_for_path (override, TRUE, NULL, error);
+    }
+
+  return flatpak_installation_new_system (NULL, error);
+}
+
+static gboolean
+deploy_flatpaks (GPtrArray *flatpaks_to_deploy, GError **error)
+{
+  g_autoptr(FlatpakInstallation) installation = get_flatpak_installation (error);
+  gsize i = 0;
+
+  if (!installation)
+    return FALSE;
+
+  for (; i < flatpaks_to_deploy->len; ++i)
+    {
+      FlatpakRemoteRef *to_deploy = g_ptr_array_index (flatpaks_to_deploy, i);
+      if (!flatpak_installation_install_full (installation,
+                                              FLATPAK_INSTALL_FLAGS_NO_PULL,
+                                              flatpak_remote_ref_get_remote_name (to_deploy),
+                                              flatpak_ref_get_kind (FLATPAK_REF (to_deploy)),
+                                              flatpak_ref_get_name (FLATPAK_REF (to_deploy)),
+                                              NULL,
+                                              NULL,
+                                              NULL,
+                                              NULL,
+                                              NULL,
+                                              NULL,
+                                              error))
+        return FALSE;
+    }
+
+  return TRUE;
+}
+
+static gboolean
+check_if_disabled_on_this_boot (gboolean *disabled, GError **error)
```

Comment:
I think you should be able to use `ConditionKernelCommandLine=!eos-updater-disable` in the unit file directly for this

Reply:
You do use the systemd condition now in the unit file, so why do you still do this manually?